SonarTfsAnnotate/HistoryProvider.cs

```diff
@@ -44,25 +45,34 @@ public HistoryProvider(IEnumerable<Changeset> changesets)
 
         public bool Next()
         {
-            if (current - 1 >= 0)
+            while (true)
             {
-                Dispose(current - 1);
-            }
+                if (current - 1 >= 0)
+                {
+                    Dispose(current - 1);
+                }
 
-            current++;
-            if (current >= changesets.Count)
-            {
-                return false;
-            }
+                current++;
+                if (current >= changesets.Count)
+                {
+                    return false;
+                }
 
-            if (current + PREFETCH_SIZE < changesets.Count)
-            {
-                Prefetch(current + PREFETCH_SIZE);
-            }
+                if (current + PREFETCH_SIZE < changesets.Count)
+                {
+                    Prefetch(current + PREFETCH_SIZE);
+                }
+
+                manualResetEvents[current].WaitOne();
 
-            manualResetEvents[current].WaitOne();
+                if (!File.Exists(filenames[current]))
+                {
+                    // The download was not successful. Move on to the next file.
+                    continue;
```

Comment:
I agree that the annotation won't be correct. Personally I'd prefer an approximation rather then no annotation at all, but I can look into switching up the behavior.

It's an invalid operation exception with the text: "TF30063: You are not authorized to access Microsoft-IIS/7.5"

It's really bizarre because it's only one revision of of one file we get the error on. It also has an odd change type: add, source rename.